perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/FacilitiesManagerBlImpl.java

```diff
@@ -329,10 +321,10 @@ public void deleteFacility(PerunSession sess, Facility facility, Boolean force)
 			for (Resource resource : resources) {
 				getPerunBl().getResourcesManagerBl().deleteResource(sess, resource);
 			}
-			List<Task> tasks = taskManager.listAllTasksForFacility(facility.getId());
+			List<Task> tasks = tasksManagerBl.listAllTasksForFacility(facility.getId());
 			for (Task task : tasks) {
-				resultManager.clearByTask(task.getId());
-				taskManager.removeTask(task.getId());
+				tasksManagerBl.clearByTask(task.getId());
```

Comment:
This is no longer human readable. Since we technically joined logic of ResultManager and TaskManager. Please rename ``clearByTask()`` method to something meaningful (at least on perun-core level, not outer API). You can do this in new commit (preferably).